### ISSUE TITLE
[Bug]: $this->paginationTotalItemCount included on all pagination types. Too heavy for "simple"

### ISSUE THREAD

**Issue by booni3** [rappasoft/laravel-livewire-tables#1687]
### What happened?

I have just upgraded to v3. 

I have some tables with nearly 1M rows and some relationships. These worked fine in V2 using simple pagination.

V3 has introduced the code:

```php
if ($this->isPaginationMethod('simple')) {

    $this->paginationTotalItemCount = $this->getBuilder()->count();

    return $this->getBuilder()->simplePaginate($this->getPerPage() === -1 ? $this->paginationTotalItemCount : $this->getPerPage(), ['*'], $this->getComputedPageName());

}
```

The `getBuilder()->count()` method is very heavy on big tables. On V2 I did not have the required indexes and tables would load in milliseconds. Without indexes on V3 my tables are timing out with > 30s load time.

After adding some additional indexes, I have gotten load times down between 500ms and 10s... but it is still a big performance from from V2 in this regard.

Do we need to be getting `$this->paginationTotalItemCount = $this->getBuilder()->count();` for simple pagination? I thought the whole idea of simple was to make things more performant?

### Package Version

3

### PHP Version

8.2.x

### Laravel Version

10.34

### Theme

None

**Comment by lrljoe** [comment]
I'm on holiday at the mo, but will look at adding a toggle for enabling/disabling the behaviour in a release when I'm back (next month)

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: Adam ***@***.***>
Sent: Thursday, March 14, 2024 7:35:44 AM
To: rappasoft/laravel-livewire-tables ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [rappasoft/laravel-livewire-tables] [Bug]: $this->paginationTotalItemCount included on all pagination types. Too heavy for "simple" (Issue #1687)


What happened?

I have just upgraded to v3.

I have some tables with nearly 1M rows and some relationships. These worked fine in V2 using simple pagination.

V3 has introduced the code:

if ($this->isPaginationMethod('simple')) {

    $this->paginationTotalItemCount = $this->getBuilder()->count();

    return $this->getBuilder()->simplePaginate($this->getPerPage() === -1 ? $this->paginationTotalItemCount : $this->getPerPage(), ['*'], $this->getComputedPageName());

}

The getBuilder()->count() method is very heavy on big tables. On V2 I did not have the required indexes and tables would load in milliseconds. Without indexes on V3 my tables are timing out with > 30s load time.

After adding some additional indexes, I have gotten load times down between 500ms and 10s... but it is still a big performance from from V2 in this regard.

Do we need to be getting $this->paginationTotalItemCount = $this->getBuilder()->count(); for simple pagination? I thought the whole idea of simple was to make things more performant?

How to reproduce the bug

No response

Package Version

3

PHP Version

8.2.x

Laravel Version

10.34

Alpine Version

No response

Theme

None

Notes

No response

Error Message

No response

**Comment by booni3** [comment]
Still interested in this.

**Comment by lrljoe** [comment]
This is fixed in PR #1755 

This does need a bit more testing before I can realistically merge it in however!  It should be released by the end of the week.

**Comment by lrljoe** [comment]
Now released as v3.2.8, please advise if anything not as expected!
